packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -519,15 +519,41 @@ describe('<DataGrid /> - Layout & Warnings', () => {
       });
     });
 
-    it('should have the correct intrinsic height when autoHeight={true}', () => {
-      render(
-        <div style={{ width: 300 }}>
-          <DataGrid {...baselineProps} headerHeight={40} rowHeight={30} autoHeight />
-        </div>,
-      );
-      expect(document.querySelector('.MuiDataGrid-main')!.clientHeight).to.equal(
-        40 + 30 * baselineProps.rows.length,
-      );
+    describe('autoHeight', () => {
+      it('should have the correct intrinsic height', () => {
+        render(
+          <div style={{ width: 300 }}>
+            <DataGrid {...baselineProps} headerHeight={40} rowHeight={30} autoHeight />
```

Comment:
We could extract `headerHeight` into its own variable for readability in the total height formula